### PR TITLE
https -> ens protocol redirect service

### PR DESCRIPTION
**Pull request by mikehelmick**
First towards #549 

## Proposed Changes

* Standalone service to redirect https:// links to ens:// for mobile 

**Release Note**

```release-note
Introduce deep link redirect service to support exposure notification express.
```

### CODE REVIEW

File: cmd/enx-redirect/main.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+	{
+		redirectController := redirect.New(ctx, config, h)
+		r.Handle("/v", redirectController.HandleIndex()).Methods("GET")
+	}
+
+	// Wrap the main router in the mutating middleware method. This cannot be
+	// inserted as middleware because gorilla processes the method before
+	// middleware.
+	mux := http.NewServeMux()
```
Comment (sethvargo) [inline cmd/enx-redirect/main.go:103]: This is probably unnecessary for this service, so no one is ever going to submit forms.

File: pkg/config/redirect_config.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/sethvargo/go-envconfig"
+)
+
+// RedirectConfig represents the environment based config for the redirect server.
+type RedirectConfig struct {
+	Observability observability.Config
+
+	Port string `env:"PORT,default=8080"`
+
+	AssetsPath string `env:"KO_DATA_PATH"`
```
Comment (sethvargo) [inline pkg/config/redirect_config.go:33]: Are we using Ko again?

File: pkg/config/redirect_config.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/google/exposure-notifications-server/pkg/observability"
+
+	"github.com/sethvargo/go-envconfig"
+)
+
+// RedirectConfig represents the environment based config for the redirect server.
+type RedirectConfig struct {
+	Observability observability.Config
+
+	Port string `env:"PORT,default=8080"`
```
Comment (sethvargo) [inline pkg/config/redirect_config.go:31]:
```suggestion
	Port string `env:"PORT, default=8080"`
```

File: pkg/controller/redirect/index.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	"strings"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+)
+
+func (c *Controller) HandleIndex() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		//ctx := r.Context()
+
+		idx := strings.Index(r.RequestURI, "/v?")
```
Comment (sethvargo) [inline pkg/controller/redirect/index.go:28]: Whats this?

File: pkg/config/redirect_config.go (hunk 2 above)
Comment (mikehelmick) [reply pkg/config/redirect_config.go:33]: just for my rapid prototype.

File: pkg/controller/redirect/index.go (hunk 4 above)
Comment (mikehelmick) [reply pkg/controller/redirect/index.go:28]: this is changed - see update.

File: cmd/enx-redirect/assets/header.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    background-color: #0055b1;
+  }
+
+  a.input-group-text:hover {
+    cursor: pointer;
+    text-decoration: none;
+  }
+
+</style>
+{{end}}
```
Comment (flagxor) [inline cmd/enx-redirect/assets/header.html:39]: line ending?

**Review by flagxor — APPROVED** [review]
lgtm
Is it overkill to have an explicit list of regions?

**Pull request merged**